dimod/constrained.py

```diff
@@ -773,9 +774,112 @@ def vartype(self, v: Variable) -> Vartype:
+        Args:
+            fp: file-like object in the LP format
+            lower_bound_default: in case lower bounds for integer are not given, this will be the default
+            upper_bound_default: in case upper bounds for integer are not given, this will be the default
 
-CQM = ConstrainedQuadraticModel
+        Returns:
+            The model encoded in the LP file, converted to ConstrainedQuadraticModel dimod object
+        """
+
+        grammar = make_lp_grammar()
+        parse_output = grammar.parseFile(fp)
+        variables_info = get_variables(parse_output, lower_bound_default, upper_bound_default)
+
+        cqm = ConstrainedQuadraticModel()
+        x = {}
+
+        # add the objective
+        obj = QuadraticModel()
+        for i, n in enumerate(variables_info):
+            if variables_info[n].type == 'b':
+                x[n] = obj.add_variable('BINARY', n)
+            elif variables_info[n].type == 'i':
+                x[n] = obj.add_variable('INTEGER', n)
+                if variables_info[n].lb is not None:
+                    obj.set_lower_bound(n, variables_info[n].lb)
+                if variables_info[n].ub is not None:
+                    obj.set_upper_bound(n, variables_info[n].ub)
+            else:
+                raise ValueError("Unknown vartype: {}".format(variables_info[n].type))
+
+        # parse and set the objective
+        obj_coefficient = 1
+        for oe in parse_output.objective:
+
+            if isinstance(oe, str):
+                if oe in obj_senses.keys():
+                    obj_coefficient = obj_senses[oe]
+
+            else:
+                if len(oe) == 2:
+                    if oe.name != "":
+                        # linear term
+                        obj.set_linear(x[oe.name[0]], obj_coefficient * oe.coef)
+
+                    else:
+                        # this is a pure squared term
+                        if variables_info[x[oe.squared_name[0]]].type == 'b':
+                            obj.set_linear(x[oe.squared_name[0]], obj_coefficient * oe.coef * 0.5)
+                        elif variables_info[x[oe.squared_name[0]]].type == "i":
+                            obj.set_quadratic(x[oe.squared_name[0]], x[oe.squared_name[0]],
+                                              obj_coefficient * oe.coef * 0.5)
+                        else:
+                            raise TypeError("Unknown variable type: {}".format(x[oe.squared_name[0]]))
+
+                elif len(oe) == 3:
+                    # bilinear term
+                    obj.set_quadratic(x[oe.name[0]], x[oe.second_var_name[0]], obj_coefficient * oe.coef * 0.5)
+
+        cqm.set_objective(obj)
 
+        # adding constraints
+        for c in parse_output.constraints:
+
+            cname = c.name[0]
+            csense = constraint_symbols[c.sense]
+            ccoef = c.rhs
+
+            # add the constraint as a list of terms. It is
+            c_model = []
```

Comment:
I suspect that you'll get a bit of a performance boost if you just construct the `QuadraticModel` here rather than a list of terms. Like you do with the objective.